### ISSUE TITLE
x/build: gitmirror or maintnerd consumed all our Gerrit quota

### ISSUE THREAD

**Issue by bradfitz** [golang/go#23853]
Maintner is blocked:

```
2018/02/15 16:25:01 gerrit go.googlesource.com/sublime-config: sync: git fetch origin: exit status 128, fatal: remote error: Daily ls-remote rate limit exceeded for IP 35.188.125.9
2018/02/15 16:25:01 IS TEMP ERROR? *errors.errorString git fetch origin: exit status 128, fatal: remote error: Daily ls-remote rate limit exceeded for IP 35.188.125.9
2018/02/15 16:25:01 Temporary error from gerrit go.googlesource.com/sublime-config: git fetch origin: exit status 128, fatal: remote error: Daily ls-remote rate limit exceeded for IP 35.188.125.9
2018/02/15 16:25:02 gerrit go.googlesource.com/term: sync: git fetch origin: exit status 128, fatal: remote error: Daily ls-remote rate limit exceeded for IP 35.188.125.9
2018/02/15 16:25:02 IS TEMP ERROR? *errors.errorString git fetch origin: exit status 128, fatal: remote error: Daily ls-remote rate limit exceeded for IP 35.188.125.9
2018/02/15 16:25:02 Temporary error from gerrit go.googlesource.com/term: git fetch origin: exit status 128, fatal: remote error: Daily ls-remote rate limit exceeded for IP 35.188.125.9
```

So now all bots are down.

GerritBot consumed all of our Gerrit quota.

@andybons, please stop GerritBot and/or increase our Gerrit quota.

**Comment by andybons** [comment]
Our ls-remote quota has been reset. After investigating with @bradfitz, the requests were coming from either mainternd or gitmirror, not GerritBot.

**Comment by bradfitz** [comment]
Sorry, I by default blame whatever changed last. I'm still debugging.

**Comment by bradfitz** [comment]
Looking at logs in the Cloud Logging console (which captures all the GKE pod output), started at "2018-02-15T13:27:16.527663505Z" , or 5:27am Pacific: "2018-02-15 05:27:14.000 PST"

I'll look at logs just before that to see if it's obvious what went crazy.

**Comment by timendez** [comment]
Is this what's causing intermittent fails with this?:
```
[ERROR]	Update failed for golang.org/x/sys: Unable to get repository: Cloning into '/root/.glide/cache/src/https-golang.org-x-sys'...
fatal: unable to access 'https://go.googlesource.com/sys/': Failed to connect to go.googlesource.com port 443: Operation timed out
: exit status 128
[ERROR]	Failed to install: Unable to get repository: Cloning into '/root/.glide/cache/src/https-golang.org-x-sys'...
fatal: unable to access 'https://go.googlesource.com/sys/': Failed to connect to go.googlesource.com port 443: Operation timed out
: exit status 128
```

**Comment by andybons** [comment]
Based on the error (presenting the IP address), we’re not authenticating our git requests somewhere. If we do this then we’ll get better error logging and much higher quota.

**Comment by andybons** [comment]
@timendez that’s unrelated to this issue.

**Comment by timendez** [comment]
@andybons dang, ok thanks

**Comment by bradfitz** [comment]
@andybons, will do. We historically never did, because we were bound to 1 IP address and got high quota for that IP. Now that we bounce around k8s nodes, I'll make them authenticate.

**Comment by bradfitz** [comment]
From another bug:

We recently upgraded our Kubernetes cluster and our 10 random container jobs re-laid themselves out onto our 4 physical nodes (i.e. 4 egress IP addresses)

We got lucky before and our 3 gerrit-hitting jobs were using different nodes (different IPs).

But after this latest upgrade, all 3 are on the same node, so we burn through that IP's quota by the end of the day.

That's the current theory.

I'm adding auth now.
